core/server/data/importer/handlers/image.js

```diff
@@ -37,8 +37,14 @@ ImageHandler = {
 
         return Promise.map(files, function (image) {
             return store.getUniqueFileName(image, image.targetDir).then(function (targetFilename) {
-                image.newPath = urlService.utils.urlJoin('/', urlService.utils.getSubdir(), urlService.utils.STATIC_IMAGE_URL_PREFIX,
-                    path.relative(config.getContentPath('images'), targetFilename));
+                image.newPath = urlService.utils.isUrlAbsolute(targetFilename) ?
+                    targetFilename :
+                    urlService.utils.urlJoin(
+                        '/',
+                        urlService.utils.getSubdir(),
+                        urlService.utils.STATIC_IMAGE_URL_PREFIX,
+                        path.relative(config.getContentPath('images'), targetFilename)
+                    );
```

Comment:
Okay I see the problem here. This piece of code is specific for the local storage adapter 😒
The adapter returns an absolute path, but we need a relative path for [replacing the original image path](https://github.com/TryGhost/Ghost/blob/1.21.1/core/server/data/importer/importers/image.js#L18).

I don't think your suggestion isn't the right solution here 🤔 

We should extend the local file storage to return either an absolute or a relative path for `getUniqueFileName`. 

(If you call `save`, the local storage already returns a relative path - so it can handle this case already)

Reply:
@iamakulov

Reply:
@kirrg001 Sure, I’ve seen this! Gimme a few days please, I’ll get to this